assets/js/script.js

```diff
@@ -0,0 +1,471 @@
+$(function() {
+
+// SCREEN BUTTONS
+  function generalStartFunctions() {
+    $gameOverlay = $('.gameOverlay')
+    $gameOverlay.css({
+      'display': 'none'
+    })
+    document.getElementsByClassName('gogo')[0].volume = 0.6;
+    document.getElementsByClassName('gogo')[0].play()
+    updatePlayerStats()
+    countDown()
+  }
+
+  $('.easyStartBtn').on('click', () => {
+    generalStartFunctions()
+    //stage specific settings
+    $gameScreen.css({
+      'background-image': 'url("./assets/img/bgAnime.jpg")'
+    })
+    document.getElementsByClassName('ghoulBGM')[0].volume = 1
+    document.getElementsByClassName('ghoulBGM')[0].play()
+    spawnIntervalEasy(4000)
+    spawnIntervalAmmo(16000)
+    startNorm(28000)
+    event5(28000)
+    speedShooting(62000)
+    event6(62000)
+    function startNorm(time) {
+      setTimeout(() => spawnIntervalNormal(6000), time)
+    }
+    function speedShooting(time) {
+      setTimeout(() => spawnIntervalEasy(2000), time)
+      setTimeout(() => spawnIntervalAmmo(10000), time)
+    }
+  })
+
+  $('.normalStartBtn').on('click', () => {
+    generalStartFunctions()
+    //stage specific settings
+    $gameScreen.css({
+      'background-image': 'url("./assets/img/nightBG.jpg")'
+    })
+    document.getElementsByClassName('believerBGM')[0].volume = 1
+    document.getElementsByClassName('believerBGM')[0].play()
+    spawnIntervalEasy(4000)
+    spawnIntervalNormal(6000)
+    spawnIntervalHard(11000)
+    spawnIntervalAlly(8000)
+    spawnIntervalAmmo(12000)
+    spawnIntervalHealth(39000)
+    event4(1000)
+    event6(15000)
+    event1(22000)
+    event1(32000)
+    event6(45000)
+    event7(54000)
+    event4(94000)
+    event3(100000)
+  })
+
+  $('.hardStartBtn').on('click', () => {
+    generalStartFunctions()
+    // stage specific settings
+    document.getElementsByClassName('deadpoolBGM')[0].currentTime = 3
+    document.getElementsByClassName('deadpoolBGM')[0].volume = 0.6
+    document.getElementsByClassName('deadpoolBGM')[0].play()
+    spawnIntervalEasy(3000)
+    spawnIntervalNormal(6000)
+    spawnIntervalHard(10000)
+    spawnIntervalAmmo(8000)
+    spawnIntervalHealth(50000)
+    spawnIntervalGrenade(50000)
+    event1(14000) //14
+    event2(35000) //35
+    event3(46000) //46
+    event2(56000) //56
+    event6(84000) //84
+    event7(116000) //116
+  })
+
+  $('.retryBtn').on('click', () => {
+    location.reload()
+  })
+
+// END OF SCREEN BUTTONS
+// GENERAL GAME FUNCTIONS ------------------------------
+
+  var playerStats = {
+    health: 5,
+    ammo: 25,
+    grenade: 3,
+  }
+  var timeCount = 120
+  var playerStatsInterval
+  var countDownInterval
+
+  function updatePlayerStats () {
+    playerStatsInterval = setInterval(() => {
+      var $playerHealth = $('.playerHealth')
```

Comment:
What will be the difference if you put these line(100 - 103) in outer scope of function `updatePlayerStats`? What will be the advantage if we reduce the number of DOM accessing?

Reply:
Ah I see, putting the variables in the set interval scope will make unnecessary DOM accessing over and over again.